content-src/components/TopSites/TopSites.js

```diff
@@ -17,14 +17,28 @@ const TopSites = React.createClass({
       <h3 className="section-title">Top Sites</h3>
       <div className="tiles-wrapper">
         {sites.map((site) => {
-          const color = site.favicon_colors && site.favicon_colors[0] && site.favicon_colors[0].color || [333, 333, 333];
+          let title;
+          let color;
+          try {
+            title = prettyHostname(site.parsedUrl.hostname);
+          } catch (e) {
+            //
+          }
+          if (!title) {
+            title = site.provider_name || site.title;
```

Comment:
Why not just move this bit up into the `let` assignment?

```js
let title = site.provider_name || site.title;

try {
  title = prettyHostname(site.parsedUrl.hostname);
} catch (e) {
 // ignore
}
```

Same with `color`, I suppose.
```js
let color = [251, 251, 251];

try {
  color = site.favicon_colors[0].color;
} catch (e) {
  // ignore
}
```